_posts/2021-12-13-log4shell-does-not-affect-trino.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title:  "Log4Shell does not affect Trino"
+author: Brian Olsen
+excerpt_separator: <!--more-->
+---
+
+In the last few days we had a surge of folks in our community reaching out with
+concerns over the [Log4Shell exploit](https://www.lunasec.io/docs/blog/log4j-zero-day/)
+([CVE-2021-44228](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228)),
+and we wanted to inform you that Trino is not affected. Trino does not use log4j
+in the core engine or runtime classes. There are some connectors that include 
+the log4j dependency from client dependencies, but some are not used or are not
+versions affected by the Log4Shell vulnerability. Rest assured, we are taking 
+extra steps to make sure no future iterations of Trino will ever introduce this 
+vulnerability.
+
+<!--more-->
+
+## Trino Connectors with the Log4j dependency
+
+If you do [a quick search in the Trino repository](https://github.com/trinodb/trino/search?p=1&q=log4j), 
+you'll notice the log4j dependency pops up in two of the connectors, Accumulo 
+and Elasticsearch.
+
+### Accumulo
+
+The Accumulo connector depends on log4j 1.2.17, which although isn't vulnerable
+to Log4Shell. Log4j v1 libraries [have other vulnerabilities](https://github.com/trinodb/trino/issues/8781)
+which do not apply to how we've used the loggers in the connector code. To be
+clear, despite the small use of this logger in the Accumulo connector, there is 
+still no threat even if you are using it. We will remove the uses of this log4j
```

Comment:
We are working on removing the dependency as a precautionary measure

and add link to github issue to track that..

Reply:
Or change “for best practices” to something like “to avoid any confusion”